src/caf/toolkit/io.py

```diff
@@ -198,15 +198,16 @@ def read_csv_matrix(
     elif format_ == "long":
         matrix = read_csv(path, index_col=kwargs.pop("index_col", [0, 1]), **kwargs)
 
-        matrix = matrix.unstack()
+        # Matrix has MultiIndex so this returns a DataFrame
+        matrix = matrix.unstack()  # type: ignore
         matrix.columns = matrix.columns.droplevel(0)
 
     else:
         raise ValueError(f"unknown format {format_}")
 
-    # Attempt to convert to integers
-    matrix.columns = pd.to_numeric(matrix.columns, errors="ignore", downcast="integer")
-    matrix.index = pd.to_numeric(matrix.index, errors="ignore", downcast="integer")
+    # Attempt to convert to integers, which should work fine for pandas Index
+    matrix.columns = pd.to_numeric(matrix.columns, errors="ignore", downcast="integer")  # type: ignore[call-overload]
```

Comment:
Just a heads up that this warning keeps appearing when running the tests:

```
 FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead
    matrix.columns = pd.to_numeric(matrix.columns, errors="ignore", downcast="integer")  # type: ignore[call-overload]
```

Is this one that's worth solving while we're here? Not sure if this would have any impact on the type error you're getting?

Reply:
Fixed